### PR TITLE
completed week  4 assessment

### PR DESCRIPTION
**Pull request by lmhazel**
# Submitting Week 4 Assessment

### ✅ Completed Files

Mark the following as confirmation that the files in this repository are complete.

- [x ] JavaScript Coding Challenges
- [ x] Ruby Coding Challenges
- [ x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

**Review by SunkissedQueen — CHANGES_REQUESTED** [review]
Lea, change request to provide the definitions for the looking ahead terms.

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,7 +13,12 @@
 // --------------------1) Create a function that takes in an array, removes the first item from the array and shuffles the remaining content.
 
 // a) Create a test with an expect statement using the variable provided.
-// HINT: Check out this resource: https://jestjs.io/docs/expect#expectarraycontainingarray
+// HINT: Check out this resource: 
+
+//Pseudo Code: 
+//Input: an array of strings
+//Output: The input array without the first item from the array and shaffles the remaining content
+//Function: A fxn called shuffledArray which uses the .shift method to first remove the first variable in the array. Next using the .sort method and a random number generator using Math.random, the remaining variables of the array will be randomly put into an order and return an array.
```
Comment (TsunamiCode5) [inline code-challenges.test.js:21]: 😍

File: code-challenges.test.js [hunk 2]
```diff
@@ -26,10 +31,28 @@ const colors2 = [
...
+//   it("returns an array where the first variable is removed and the remaining variables' order is shuffled", () => {
+//     expect(["yellow", "blue", "pink", "green"]).toEqual(expect.arrayContaining(expected));
+//   });
+//     it('does not match if received does not contain expected elements', () => {
+//       expect(["yellow", "blue", "pink"]).not.toEqual(expect.arrayContaining(expected));
+//   })
+// })
 
 // b) Create the function that makes the test pass.
+// const shuffledArray = (array) => {
```
Comment (TsunamiCode5) [inline code-challenges.test.js:45]: Nice approach!

File: code-challenges.test.js [hunk 3]
```diff
@@ -38,7 +61,16 @@ const votes1 = { upVotes: 13, downVotes: 2 }
 const votes2 = { upVotes: 2, downVotes: 33 }
 // Expected output: -31
 
+// describe("endTally", () => {
+//   it("returns the up votes and down votes end tally", () => {
+//     expect(greeter()).toEqual(11)
```
Comment (TsunamiCode5) [inline code-challenges.test.js:66]: greeter?

File: ruby-challenges.rb [hunk 4]
```diff
@@ -9,8 +9,14 @@
 # Expected output: '42 is even'
 num3 = 221
 # Expected output: '221 is odd'
-
-
+def number_checker num
```
Comment (TsunamiCode5) [inline ruby-challenges.rb:12]: Nice!

File: ruby-challenges.rb [hunk 5]
```diff
@@ -21,7 +27,10 @@
 # Expected output: 'Sgt Pppr'
 beatles_album3 = 'Abbey Road'
 # Expected output: 'bby Rd'
-
+def vowel_remover string
```
Comment (TsunamiCode5) [inline ruby-challenges.rb:30]: Crushed it.

**Review by TsunamiCode5 — APPROVED** [review]
Lea, you did a really good job taking the tips from other assessments and implementing them in this one. Your pseudocode has more details and great vocabulary terms incorporated, your code has really good logic, and you're doing really well elaborating on talking points in your interview questions. This assessment comprehension level is advanced beginner. You can easily level this up with The Dreyfus model describing competent level as detailed pseudo code with proper vocabulary terms, "your answers" written in complete sentences the student's own words that provide correct answers, and well researched and detailed answers written in complete sentences in the student's own words that provides additional context. Great work!

**Pull request merged**